esmvalcore/preprocessor/_regrid.py

```diff
@@ -563,105 +742,30 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
-        ycoord = target_grid.coord(axis='y', dim_coords=True)
-        xcoord.coord_system = src_cs
-        ycoord.coord_system = src_cs
-    elif isinstance(target_grid, dict):
-        # Generate a target grid from the provided specification,
-        target_grid = _regional_stock_cube(target_grid)
-
-    if not isinstance(target_grid, iris.cube.Cube):
-        raise ValueError(f'Expecting a cube, got {target_grid}.')
-
-    if isinstance(scheme, dict):
-        scheme = dict(scheme)  # do not overwrite original scheme
-        try:
-            object_ref = scheme.pop("reference")
-        except KeyError as key_err:
-            raise ValueError(
-                "No reference specified for generic regridding.") from key_err
-        module_name, separator, scheme_name = object_ref.partition(":")
-        try:
-            obj = importlib.import_module(module_name)
-        except ImportError as import_err:
-            raise ValueError(
-                "Could not import specified generic regridding module. "
-                "Please double check spelling and that the required module is "
-                "installed.") from import_err
-        if separator:
-            for attr in scheme_name.split('.'):
-                obj = getattr(obj, attr)
-
-        scheme_args = inspect.getfullargspec(obj).args
-        # Add source and target cubes as arguments if required
-        if 'src_cube' in scheme_args:
-            scheme['src_cube'] = cube
-        if 'grid_cube' in scheme_args:
-            scheme['grid_cube'] = target_grid
-
-        loaded_scheme = obj(**scheme)
-    else:
-        loaded_scheme = HORIZONTAL_SCHEMES.get(scheme.lower())
-
-    if loaded_scheme is None:
-        emsg = 'Unknown regridding scheme, got {!r}.'
-        raise ValueError(emsg.format(scheme))
-
-    # Unstructured regridding requires x2 2d spatial coordinates,
-    # so ensure to purge any 1d native spatial dimension coordinates
-    # for the regridder.
-    if scheme == 'unstructured_nearest':
-        for axis in ['x', 'y']:
-            coords = cube.coords(axis=axis, dim_coords=True)
-            if coords:
-                [coord] = coords
-                cube.remove_coord(coord)
+    # Load target grid and select appropriate scheme
+    target_grid_cube = _get_target_grid_cube(
+        cube, target_grid, lat_offset=lat_offset, lon_offset=lon_offset,
+    )
 
     # Horizontal grids from source and target (almost) match
     # -> Return source cube with target coordinates
-    if _horizontal_grid_is_close(cube, target_grid):
+    if _horizontal_grid_is_close(cube, target_grid_cube):
         for coord in ['latitude', 'longitude']:
-            cube.coord(coord).points = target_grid.coord(coord).points
-            cube.coord(coord).bounds = target_grid.coord(coord).bounds
+            cube.coord(coord).points = target_grid_cube.coord(coord).points
+            cube.coord(coord).bounds = target_grid_cube.coord(coord).bounds
```

Comment:
Not new in the pull request, but maybe it would be nice not to realize the coordinates here?
```suggestion
            cube.coord(coord).points = target_grid_cube.coord(coord).core_points()
            cube.coord(coord).bounds = target_grid_cube.coord(coord).core_bounds()
```
and what abound the bounds, why are they not updated?

Reply:
But from the code it looks like the bounds are update, aren't they?

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2231/commits/cdb1f47e5296893c97003043f403e7dcb2a1e3a8